--- src/openrct2/platform/macos.m.orig	2017-12-27 11:51:38.000000000 -0800
+++ src/openrct2/platform/macos.m	2017-12-27 11:51:49.000000000 -0800
@@ -16,8 +16,8 @@
 
 #if defined(__APPLE__) && defined(__MACH__)
 
-@import AppKit;
-@import Foundation;
+#include <AppKit/AppKit.h>
+#include <Foundation/Foundation.h>
 #include <mach-o/dyld.h>
 #include "platform.h"
 #include "../util/util.h"
@@ -26,11 +26,13 @@
 
 void macos_disallow_automatic_window_tabbing()
 {
+#if __MAC_OS_X_VERSION_MAX_ALLOWED >= 101200
     @autoreleasepool {
         if ([NSWindow respondsToSelector:@selector(setAllowsAutomaticWindowTabbing:)]) {
             [NSWindow setAllowsAutomaticWindowTabbing:NO];
         }
     }
+#endif
 }
 
 void platform_get_exe_path(utf8 *outPath, size_t outSize)
@@ -155,15 +157,6 @@
         }
 
 
-        // Only check for a matching language
-        NSString *languageCode = [[preferredLocale componentsSeparatedByString:@"-"] firstObject];
-        for (int i = 1; i < LANGUAGE_COUNT; i++) {
-            NSString *optionLanguageCode = [[[NSString stringWithUTF8String:LanguagesDescriptors[i].locale] componentsSeparatedByString:@"-"] firstObject];
-            if([languageCode isEqualToString:optionLanguageCode]) {
-                *languageIdentifier = i;
-                return YES;
-            }
-        }
 
         return NO;
     }
@@ -173,7 +166,7 @@
 {
     @autoreleasepool
     {
-        NSArray<NSString*> *preferredLanguages = [NSLocale preferredLanguages];
+        NSArray *preferredLanguages = [NSLocale preferredLanguages];
         for (NSString *preferredLanguage in preferredLanguages) {
             uint16 languageIdentifier;
             if (platform_has_matching_language(preferredLanguage, &languageIdentifier)) {
