--- src/openrct2/common.h.orig	2017-12-27 11:00:50.000000000 -0800
+++ src/openrct2/common.h	2017-12-27 11:01:41.000000000 -0800
@@ -224,7 +224,12 @@
     #define RESTRICT
 #endif
 
+#ifdef __cplusplus
 #define assert_struct_size(x, y) static_assert(sizeof(x) == (y), "Improper struct size")
+#else
+#define static_assert(x,y) _Static_assert(x,y)
+#define assert_struct_size(x, y) _Static_assert(sizeof(x) == (y), "Improper struct size")
+#endif
 
 #ifdef PLATFORM_X86
     #ifndef FASTCALL
