--- m4/boost.m4.orig	2018-09-02 14:23:04.000000000 -0700
+++ m4/boost.m4	2018-09-02 14:23:29.000000000 -0700
@@ -403,7 +403,7 @@
       LDFLAGS=$boost_save_LDFLAGS
       LIBS=$boost_save_LIBS
       if test x"$Boost_lib" = xyes; then
-        Boost_lib_LDFLAGS="-L$boost_ldpath -Wl,-R$boost_ldpath"
+        Boost_lib_LDFLAGS="-L$boost_ldpath"
         Boost_lib_LDPATH="$boost_ldpath"
         break 6
       else
