--- configure.old	2019-04-14 16:32:23.000000000 -0700
+++ configure	2019-04-14 16:33:29.000000000 -0700
@@ -265,7 +265,7 @@
 
     sdk=$(getSingleQMakeVariable "QMAKE_MAC_SDK" "$1")
     if [ -z "$sdk" ]; then echo "QMAKE_MAC_SDK must be set when building on Mac" >&2; exit 1; fi
-    sysroot=$(/usr/bin/xcodebuild -sdk $sdk -version Path 2>/dev/null)
+    sysroot="/"
     if [ -z "$sysroot" ]; then echo "Failed to resolve SDK path for '$sdk'" >&2; exit 1; fi
 
     case "$sdk" in
