From afbfe1b92714f649c84fce2a663dd0b69fd30a25 Mon Sep 17 00:00:00 2001
From: Riccardo Mottola <rm@gnu.org>
Date: Wed, 4 Jul 2018 08:17:37 +0200
Subject: [PATCH] fix register structure access on x86

---
 js/src/asmjs/AsmJSSignalHandlers.cpp | 14 +++++++++++---
 1 file changed, 11 insertions(+), 3 deletions(-)

diff --git a/js/src/asmjs/AsmJSSignalHandlers.cpp b/js/src/asmjs/AsmJSSignalHandlers.cpp
index 4b14cc4c8..62defb7c3 100644
--- a/js/src/asmjs/AsmJSSignalHandlers.cpp
+++ b/js/src/asmjs/AsmJSSignalHandlers.cpp
@@ -26,6 +26,14 @@
 #include "jit/Disassembler.h"
 #include "vm/Runtime.h"
 
+#ifdef XP_MACOSX
+#ifndef __ppc__
+#include <sys/ucontext.h>
+#include <mach/mach_types.h>
+#include <mach/thread_status.h>
+#endif
+#endif
+
 using namespace js;
 using namespace js::jit;
 
@@ -202,9 +210,9 @@ class AutoSetHandlingSignal
 #  define R15_sig(p) ((p)->uc_mcontext.mc_r15)
 # endif
 #elif defined(XP_DARWIN)
-# define EIP_sig(p) ((p)->uc_mcontext->__ss.__eip)
-# define RIP_sig(p) ((p)->uc_mcontext->__ss.__rip)
-# define R15_sig(p) ((p)->uc_mcontext->__ss.__pc)
+# define EIP_sig(p) ((p)->uc_mcontext->ss.eip)
+# define RIP_sig(p) ((p)->uc_mcontext->ss.rip)
+# define R15_sig(p) ((p)->uc_mcontext->ss.pc)
 #else
 # error "Don't know how to read/write to the thread state via the mcontext_t."
 #endif
